pkg/apis/core/v1alpha1/clusteroperation.go

```diff
@@ -0,0 +1,92 @@
+	ClusterOperationResourceName = "clusteroperation"
+	// ClusterOperationResourcePlural is plural for "clusteroperation" resource
+	ClusterOperationResourcePlural = "clusteroperations"
+	// ClusterOperationShortName  is the shortname for "clusteroperation" resource
+	ClusterOperationShortName = "cor"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +k8s:deepcopy-gen=true
+
+// ClusterOperation represents a ClusterOperation resource
+type ClusterOperation struct {
+	meta_v1.TypeMeta   `json:",inline"`
+	meta_v1.ObjectMeta `json:"metadata"`
+	Spec               ClusterOperationSpec   `json:"spec"`
+	Status             ClusterOperationStatus `json:"status,omitempty"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ClusterOperationList is a list of ClusterOperation
+type ClusterOperationList struct {
+	meta_v1.TypeMeta `json:",inline"`
+	meta_v1.ListMeta `json:"metadata"`
+	Items            []ClusterOperation `json:"items"`
+}
+
+// ClusterOperationSpec is the spec used to define a ClusterOperation
+type ClusterOperationSpec struct {
+	PoolSelector ClusterOperationPoolSelector `json:"poolSelector"`
+	Operation    string                       `json:"operation"`
+	Params       ClusterOperationParams       `json:"params"`
+}
+
+// ClusterOperationStatus is the status of a cluster operation resource
+type ClusterOperationStatus struct {
+	Stage  ClusterOperationStageType `json:"stage"`
+	Status string                    `json:"status"`
+}
+
+// ClusterOperationParams parameters of cluster operation
+type ClusterOperationParams struct {
+	NewSize         string `json:"newSize"`
+	ScaleFactor     string `json:"scaleFactor"`
+	ResizeOperation string `json:"resizeOperation"`
+}
+
+// ClusterOperationPoolSelector is a pool selector description
+type ClusterOperationPoolSelector struct {
+	//TODO: define the structure
+}
+
+// ClusterOperationStatusType is types of statuses of a group snapshot operation
+type ClusterOperationStatusType string
+
+const (
+	// ClusterOperationStatusInitial //TODO
+	ClusterOperationStatusInitial ClusterOperationStatusType = ""
+	// ClusterOperationStatusPending //TODO
+	ClusterOperationStatusPending ClusterOperationStatusType = "Pending"
+	// ClusterOperationStatusInProgress //TODO
+	ClusterOperationStatusInProgress ClusterOperationStatusType = "InProgress"
+	// ClusterOperationStatusFailed //TODO
+	ClusterOperationStatusFailed ClusterOperationStatusType = "Failed"
+	// ClusterOperationStatusSuccessful //TODO
+	ClusterOperationStatusSuccessful ClusterOperationStatusType = "Successful"
+)
+
+// ClusterOperationStageType is the stage of the group snapshot
+type ClusterOperationStageType string
+
+const ( //TODO: Stolen from storagenodecluster statuses. Define own stages
+	// ClusterOperationStageInitial is when the cluster operation is just created
+	ClusterOperationStageInitial ClusterOperationStageType = ""
```

Comment:
Need an active stage which will be used when the cluster operation is actually being performed.

Reply:
I left this part for the time, when I understand stages and statuses better. Will fix it shortly, just not ready right now